covbot.py

```diff
@@ -321,10 +326,269 @@ def _get_data_for(self, query: str) -> list:
 
         return []
 
+    def _short_location(self, location: str, length=int(12)) -> str:
+        """Returns a shortened location name.
+
+        If exactly matches a country code, return that. (1)
+
+        If shorter/equal than length, return intact. (2)
+
+        Logic done in that order so that if someone passes a list
+        of countries, they get the codes back, rather than a mix
+        of codes and country names.
+
+        If longer, split on commas and replace the final part with
+        a country code if that matches. (3)
+
+        If still too long, strip out 'middle' to get desired length.
+
+        TODO: - consider stripping out ", City of,"
+              - consider simple truncation
+
+        Example (length=12):
+            United States → US
+            Manchester, GB → Manch..r ,GB
+        """
+
+        # Exact country case (1)
+        try:
+            return pycountry.countries.lookup(location).alpha_2
+        except LookupError:
+            pass
+
+        # It fits already (2)
+        if len(location) <= length:
+            return location
+
+        # If there's commas, try to replace the last bit with a
+        # country code (3)
+        if "," in location:
+            loc_parts = [s.strip() for s in location.split(",")]
+            if pycountry.countries.lookup(loc_parts[-1]):
+                loc_parts[-1] = pycountry.countries.lookup(
+                    loc_parts[-1]).alpha_2
+            location = " ,".join(loc_parts)
+
+        # If what we have is still longer, cut out the middle (4)
+        if len(location) <= length:
+            return location
+        else:
+            return "..".join([
+                location[:int((length-2)/2)],
+                location[-int((length-2)/2):]
+            ])
+
+    async def _get_multiple_locations(self, event: MessageEvent, location: str) -> dict:
+        """Split locations on ';' and look up"""
+
+        results = {}
+
+        if ";" in location:
+            locs = location.split(";")
+            for loc in locs:
+                self.log.info(f"Looking up {loc}")
+                matches = await asyncio.get_running_loop().run_in_executor(
+                    None, self._get_data_for, loc)
+                if len(matches) == 0:
+                    await self._respond(event,
+                                        f"I cannot find a match for {loc}")
```

Comment:
Ideally these would match the `!cases` 'not found' / 'multiple results' text.

That would be a simple case of reusing the strings, but I didn't wan to alter code outside of the remit of the table commands without prior approval.

Reply:
Np, I'll sort it out after the merge :+1: .